src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs

```diff
@@ -659,6 +659,32 @@ protected bool HasExplicitAccessModifier
         internal bool IsAutoProperty
             => IsSetOnEitherPart(Flags.HasAutoPropertyGet | Flags.HasAutoPropertySet);
 
+        internal bool HasAutoPropertyGet
+            => IsSetOnEitherPart(Flags.HasAutoPropertyGet);
+
+        internal bool HasAutoPropertySet
+            => IsSetOnEitherPart(Flags.HasAutoPropertySet);
+
+        /// <summary>
+        /// True if the property has a synthesized backing field, and
+        /// either no accessor or the accessor is auto-implemented.
+        /// </summary>
+        internal bool CanUseBackingFieldDirectlyInConstructor(bool useAsLvalue)
+        {
+            if (BackingField is null)
+            {
+                return false;
+            }
+            if (useAsLvalue)
```

Comment:
Will this work for a `ref`-returning property?

Reply:
Actually, I suppose it would work fine, since you can't use the backing field directly in that case... Nevermind.

Reply:
Auto-properties cannot be `ref`-returning.